tests/python/test_spark/test_spark_local.py

```diff
@@ -41,6 +40,322 @@
 pytestmark = testing.timeout(60)
 
 
+@pytest.fixture
+def spark() -> Generator[SparkSession, None, None]:
```

Comment:
If that's the case, then the existing logic for creating test sessions is also problematic. Is there a solution?

Reply:
NVM, I see there's a call to `stop`

Reply:
Fixed, scope is limited to each class and `stop` is called during tear down.

Reply:
The original design is ok, since pytest is running one by one, and once test-spark-local is finished, the spark will be stopped.

Have you tested if "scope" is working?